### ISSUE TITLE
поддержка чайника M170S

### ISSUE THREAD

**Issue by oleg77s** [Andrey-m/homebridge-redmond-sh#2]
Можно попросить адаптировать плагин для чайника M170S ?
вот guinmoon отредактировал  для чайника M170S  скрипт connect.sh:
https://github.com/PimenovAlexander/r4s-bluetooth/issues/5#issuecomment-449617043
спасибо

**Comment by Andrey-m** [comment]
Просто возьмите исправленный скрипт
и в конфиге homebridge добавьте
`{
                "name": "Чайник",
                "on_cmd": "/home/pi/connect.sh 00:00:00:00:00:00 keeptemp 95",
                "off_cmd": "/home/pi/off.sh 00:00:00:00:00:00 off",
                "state_cmd": "/home/pi/status.sh 00:00:00:00:00:00 | grep -l 'ON'",
                "manufacturer": "Redmond",
                "model": "M170S"
}`

должно помочь.
У вас появится два чайника. один будет как включить/выключить, второй на нагрев

**Comment by oleg77s** [comment]
рабочий только такой конфиг:
{
                    "name": "RK-M170S_нагрев_95",
                    "on_cmd": "sudo /home/pi/connect1.sh C9:8C:39:B6:29:E4 keeptemp 95",  (или on)
                    "off_cmd": "/home/pi/off.sh C9:8C:39:B6:29:E4",
                    "state_cmd": "/home/pi/status.sh C9:8C:39:B6:29:E4 | grep -l 'ON'",
                    "manufacturer": "Redmond",
                    "model": "RK-M170S"
                }
-----------------------
"on_cmd": " /home/pi/connect1.sh C9:8C:39:B6:29:E4 keeptemp 95"
"on_cmd": " /home/pi/connect1.sh C9:8C:39:B6:29:E4 on"
"on_cmd": " /home/pi/on.sh C9:8C:39:B6:29:E4" 
 чайник не включается
-------------------------
два чайника не удалось сделать:
"on_cmd": " /home/pi/connect1.sh C9:8C:39:B6:29:E4 keeptemp 95"
второй недоступен в доме
или оба недоступны
[2019-2-4 17:43:23] [Switches] Failed to determine RK-M170S_нагрев_95 state.
[2019-2-4 17:43:23] [Switches] connect: Device or resource busy (16)
[2019-2-4 17:43:38] [Switches] Failed to determine RK-M170S_on state.
[2019-2-4 17:43:38] [Switches] connect: Device or resource busy (16)

конфиг

"name": "Switches",
            "switches": [
                {
                    "name": "RK-M170S_нагрев_95",
                    "on_cmd": "sudo /home/pi/connect1.sh C9:8C:39:B6:29:E4 keeptemp 95",
                    "off_cmd": "/home/pi/off.sh C9:8C:39:B6:29:E4",
                    "state_cmd": "/home/pi/status.sh C9:8C:39:B6:29:E4 | grep -l 'ON'",
                    "manufacturer": "Redmond",
                    "model": "RK-M170S"
                },
                {
                    "name": "RK-M170S_on",
                    "on_cmd": "sudo /home/pi/connect1.sh C9:8C:39:B6:29:E4 on",
                    "off_cmd": "/home/pi/off.sh C9:8C:39:B6:29:E4",
                    "state_cmd": "/home/pi/status.sh C9:8C:39:B6:29:E4 | grep -l 'ON'",
                    "manufacturer": "Redmond",
                    "model": "M170S"
                }
            ]
        },

**Comment by Andrey-m** [comment]
Да. С двумя чайниками по одному адресу точно будут проблемы. Homebridge опрашивает их не в порядке очереди, а все сразу.

Скажите, а чайник включается, если в конфиге оставить только конфигурацию keeptemp?

`{
"name": "RK-M170S_нагрев_95",
"on_cmd": "sudo /home/pi/connect1.sh C9:8C:39:B6:29:E4 keeptemp 95",
"off_cmd": "/home/pi/off.sh C9:8C:39:B6:29:E4",
"state_cmd": "/home/pi/status.sh C9:8C:39:B6:29:E4 | grep -l 'ON'",
"manufacturer": "Redmond",
"model": "RK-M170S"
}`

**Comment by oleg77s** [comment]
"on_cmd": "sudo /home/pi/connect1.sh C9:8C:39:B6:29:E4 keeptemp 95" это работает,
а из терминала и без sudo включается

**Comment by Andrey-m** [comment]
Я запутался. В чем собственно проблема? Если чайник включается в нужный режим нагрева.

**Comment by oleg77s** [comment]
проблемы уже нет :)
вопрос в другом, можно ли из из модернизированного скрипта connect1 сделать новый рабочий скрипт on.sh ?
